### PR TITLE
chore :  Add  OpenLibertyGeneratorIsApplicableTest in `jkube-kit-openliberty` + openliberty legacy syntax quickstart (#1150)

### LINKED ISSUE #1150: KubernetesBuildTask: k8sBuild Generator - OpenLiberty
## Description

Child of #31 

k8sBuild Gradle task works with OpenLiberty Generator.

- Any configuration setting should be part of the KubernetesExtension class (Tasks should have no Inputs)
- Any logic present in the current Mojos should be moved to JKube-kit

## AC
- [ ] Generator configuration options can be overridden via DSL
- [ ] Documentation related to the generator in Gradle docs

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description
Fix #1150 #1151
+ test for OpenLibertyGenerator should be applicable when OpenLiberty Gradle Plugin is detected
+ Add OpenLiberty Quickstart for Gradle with deprecated plugin sytax

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [X] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [X] I have implemented unit tests to cover my changes
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [X] I tested my code in Kubernetes
 - [X] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
The buildscript dependencies needs a lot of explanation. Maybe it's better to walk through this one together.
**Our** shouldn't rely on any of the buildscript dependencies.

**Comment by rohanKanojia** [comment]
@manusa : Shall I remove code related to detecting buildscript dependencies? When generating a project from OpenLiberty starter I see Liberty Gradle Plugin configured via buildscript instead of plugins section

**Comment by manusa** [comment]
> @manusa : Shall I remove code related to detecting buildscript dependencies? When generating a project from OpenLiberty starter I see Liberty Gradle Plugin configured via buildscript instead of plugins section

Those dependencies **only** apply to the code present in the build.gradle file.

> When generating a project from OpenLiberty starter I see Liberty Gradle Plugin configured via buildscript instead of plugins section

I'm not sure where you are seeing that. e.g. in your quickstart, line 1 `apply plugin: 'liberty'` is the one responsible for the plugin configuration. In the same fashion, the dependencies you declare for our plugin in the buildscript section, shouldn't be there.

As I said, maybe it's better if we review this together.

File: gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/GradleUtilTest.java [hunk 1]
```diff
@@ -163,6 +168,25 @@ public void extractDependencies_withMultipleAndDuplicateDependencies_shouldRetur
         );
   }
 
+  @Test
+  public void extractDependency_withDependencyInBuildScript_shouldReturnDependencies() {
+    // Given
+    ScriptHandler scriptHandler = createMockedBuildScript();
```
Comment (sunix) [inline gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/GradleUtilTest.java:174]: maybe rename it `mockedBuiltScript` ?

File: gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/GradleUtilTest.java [hunk 2]
```diff
@@ -335,6 +359,31 @@ public void canBeResolved_DeprecatedAndNullResolutionAlternativesAndResolvable_s
     };
   }
 
+  private ScriptHandler createMockedBuildScript() {
+    final ModuleComponentIdentifier moduleComponentIdentifier = mock(ModuleComponentIdentifier.class);
```
Comment (sunix) [inline gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/GradleUtilTest.java:363]: This part is hard to read, there is a lot nested mocks. Could this be simplified with `Mockito.RETURNS_DEEP_STUBS` ?

Comment (rohanKanojia) [reply gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/GradleUtilTest.java:363]: Umm, do you mean using something like this?
```java
  private ScriptHandler mockedBuildScript() {
    return mock(ScriptHandler.class, RETURNS_DEEP_STUBS);
  }
```

I tried this but test is failing after this change. I'm getting this error:
```
java.lang.NullPointerException: plugins cannot be null
```

**Comment by rohanKanojia** [comment]
I checked with OpenLiberty Gradle Plugin maintainers and they don't  see any issue in checking both buildscript and plugins : https://github.com/OpenLiberty/ci.gradle/issues/695#issuecomment-1015808947

**Comment by manusa** [comment]
As discussed in internal chat, let's sync on this topic later today.

**Review by manusa — APPROVED** [review]

**Review by sunix — APPROVED** [review]

**Pull request merged**